cmd/ddev/cmd/config.go

```diff
@@ -340,6 +340,12 @@ func getConfigApp(_ string) *ddevapp.DdevApp {
 	if err != nil {
 		util.Failed("Could not create a new config: %v", err)
 	}
+	// If the project name is changed to something else in the config overrides, update it here.
+	appWithOverrides, err := ddevapp.NewApp(appRoot, true)
+	if err != nil {
+		util.Failed("Could not create a new config: %v", err)
+	}
+	app.Name = appWithOverrides.Name
```

Comment:
Why just do the name? Isn't this a general problem with anything that might be in the override file? Like PHP version, etc?

Reply:
Because it will bring all overrides that you have in `.ddev/config.local.yaml` to `.ddev/config.yaml`.

If you have something like this in the override:

```yaml
additional_hostnames:
    - foo
    - bar
```

it will be copied to the main config file. Do we really want that?

Reply:
I mean, people use these overrides to keep the main config clean, and if you pull some add-on, run `ddev config --auto` and remove that add-on, the entire add-on config (defined in `config.*.yaml`) will still be in your main config.

Reply:
Agreed. This is a difficult problem, and the long-term solution is probably in the other issue. Here are you rewriting the `name` in config.yaml or leaving it alone in the end?

Probably we should be only writing things that originally exist in the config.yaml and aren't overwritten by `config.*.yaml` That's probably difficult. OK with smaller scope, but hate to solve only one possible problem.

We may want to just say to people using `config.*.yaml` "do not use `ddev config`"

Reply:
Or maybe `ddev config` should just detect the existence of `config.*.yaml` and refuse, saying "Please edit your files instead of using this"

Reply:
> Here are you rewriting the name in config.yaml or leaving it alone in the end?

`name` is written all the time when you run `ddev config`.

> Or maybe `ddev config` should just detect the existence of `config.*.yaml` and refuse, saying "Please edit your files instead of using this"

I think this will break a lot of automated workflows.